File: c++/Palindrome.cpp
#include<iostream>
using namespace std;

int main(){
	int digit,num,n,rev;
	cout<<"Enter a number = ";
	cin>>num;
	n=num;
	while(num!=0){
		digit= num%10;
		rev=(rev*10)+digit;
		num= num/10;
	}
	if(n==rev){
		cout<<"palindrome"<<endl;
	}
	else{
		cout<<"not a palindrome"<<endl;
	}
	return 0;
}

File: java/palindrome.java
//Documentation (Write a java program to check if number is palindrome or not)
//Package
//Import
//Interface
import java.util.Scanner;
class Palindrome
{
    public static void main(String args[])
    {
        int num, temp, rem, sum=0;
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter a Number:-  ");
        num = sc.nextInt();
        temp = num;
        while (num > 0)
        {
            rem = num%10;
            sum = (sum * 10) + rem;
            num = num / 10;
        }
        if(temp == sum)
            System.out.println("The number is Palindrome");
        else
            System.out.println("The number is not Palindrome");
    }
}
